### ISSUE TITLE
RP2040 Keyboard library related shenanigans

### ISSUE THREAD

**Issue by SeongGino** [SeongGino/ir-light-gun-plus#4]
- We can't use the main Arduino keyboard library with RP2040, as Earle Philhower's core on versions above 3.0.0 outright deprecated and conflicts with it.
- A replacement library has been tentatively included in GUN4ALL for RP2040, but while it works it adds another keyboard device and makes the original one non-functioning. Perhaps we should merge both mouse and keyboard libraries together? That might require some tweaks to the LightgunButtons library as well.
- Replacement library also does its own delays with every keypress report, hopefully maybe making keyboard presses more reliable?

**Comment by SeongGino** [comment]
I've decided I'll work to merge the 'new' keyboard library with AbsMouse5, and have both integrated as the primary combined input library.

Using both in one library should simplify the use for not just RP2040 but potentially other boards - but since it applies its own small delay for presses it *might* also fix the sticking buttons issue with keyboard inputs that I've worked around.

**Comment by SeongGino** [comment]
dbe06d46a63e15d98005e6128e9fea68c536ba7c through c1f3d9235705ba47b23bc778452ab49b98af9ee4 resolves this issue.

As a result, TinyUSB is now somewhat of a hard requirement, as we no longer have a normal HID-compatible Keyboard implementation. It's only a small setback, as all endorsed microcontrollers support the TinyUSB stack (and I've confirmed DOES build without issues); and in exchange, this fixes input handling (in non-serial handoff mode) so we can rely more on LightgunButtons as we should.

Closing.
